src/components/search/SearchDirective.js

```diff
@@ -183,10 +176,42 @@ goog.require('ga_search_type_directives');
 
       $scope.layers = $scope.map.getLayers().getArray();
 
+      $scope.setKeyboardNav = function(isNav) {
+        keyboardNav = isNav;
+      };
+
+      var losingFocus = false;
+      $scope.lostFocus = function() {
+        if (!keyboardNav) {
+          // We apply a timeout of 200 ms to make
+          // sure the click event on the result item is fired
+          // before the blur event on the input
+          losingFocus = true;
+          $timeout(function() {
+            if (losingFocus) {
+              $scope.searchFocused = false;
+              losingFocus = false;
+            }
+          }, 200);
+        }
+      };
+
+      $scope.onFocus = function() {
+        losingFocus = false;
+        $scope.setKeyboardNav(false);
+        $scope.searchFocused = true;
+      };
+
+      $scope.showDropDown = function() {
+        var hasResults = restat.sets() > 0 ? true : false;
```

Comment:
Isn't `restat.sets()` enough? Should be truthy when there are results and falsy when not.